crates/re_renderer/src/resource_managers/texture_manager.rs

```diff
@@ -346,52 +358,79 @@ impl TextureManager2D {
     ) -> Result<GpuTexture2D, TextureCreationError> {
         re_tracing::profile_function!();
 
-        if creation_desc.width == 0 || creation_desc.height == 0 {
-            return Err(TextureCreationError::ZeroSize(creation_desc.label.clone()));
+        let Texture2DCreationDesc {
+            label,
+            data,
+            format,
+            width,
+            height,
+        } = creation_desc;
+        let (width, height, format) = (*width, *height, *format);
+
+        if width == 0 || height == 0 {
+            return Err(TextureCreationError::ZeroSize(label.clone()));
         }
 
         let max_texture_dimension_2d = device.limits().max_texture_dimension_2d;
-        if creation_desc.width > max_texture_dimension_2d
-            || creation_desc.height > max_texture_dimension_2d
-        {
+        if width > max_texture_dimension_2d || height > max_texture_dimension_2d {
             return Err(TextureCreationError::TooLarge {
-                width: creation_desc.width,
-                height: creation_desc.height,
+                width,
+                height,
                 max_texture_dimension_2d,
             });
         }
 
+        if !format.is_compressed() {
```

Comment:
Why not? Because nothing in this function uses `Texture2DBufferInfo`, so it didn't occur to me. Also it requires me to write more code and understand what `buffer_size_unpadded` is in relation to the input data.

Counter-question: _why_ use `Texture2DBufferInfo` and `buffer_size_unpadded` here? Would it make the code shorter? Or more correct?

Reply:
it sure would make it more correct. As is the code will fail again in the future when somebody comes with a compressed format in which case we'll don't do any check at all then

Reply:
> Because nothing in this function uses Texture2DBufferInfo, so it didn't occur to me.

it probably should though, the function predates this utility :)

Reply:
I mean it's not a biggy, I approved the PR didn't I? 😄